docs/design/coreclr/botr/clr-abi.md

```diff
@@ -122,6 +130,8 @@ Primitive value types smaller than 32-bits are widened to 32-bits: signed small
 
 Small primitive arguments have undefined upper bits. This can be different from the standard calling conventions that may require normalization (e.g. on ARM32 and Apple ARM64).
 
+On RISC-V small primitive arguments are extended according to standard calling conventions.
```

Comment:
The managed calling convention leaves the upper bits of small primitive arguments undefined across all architectures. It does not match the native calling convention wrt. passing small types on number of other architectures.

Is there a value for RISC-V to be special on this point?

Reply:
It's natural for RISC-V to have small primitives extended, any 32-bit instruction leaves the register extended so it's cheaper to require that of the caller (it's free in most cases) than to do 2 shifts for each argument every time in the callee, which we couldn't optimize out otherwise. Some care was already taken to properly extend small integers, e.g. in https://github.com/dotnet/runtime/pull/93665#discussion_r1365353888 so we're almost there anyway.

Is there a benefit of unifying it given each architecture strives to follow their own platform calling conventions?

Reply:
Ok, sounds reasonable to me. It is another one of those things that make RISCV different.